### PR TITLE
require node_name option

### PR DESCRIPTION
**Pull request by mitchellhenke**

### CODE REVIEW

**Comment by jeregrine** [comment]
Lgtm

**Comment by hamiltop** [comment]
I solved this on my system by providing `--sname foo` as an argument when launching. This was easy because each node has a different hostname, so `foo@bar`, `foo@baz`, etc. are all unique.

The solution here seems a bit more cumbersome.  I guess I could just do `System.get_env("HOSTNAME")` to accomplish the same effect, but if the BEAM node name is set, we shouldn't have to specify it again here.

File: lib/phoenix_pubsub_redis/redis.ex [hunk 1]
```diff
@@ -59,7 +60,7 @@ defmodule Phoenix.PubSub.Redis do
     pool_name   = Module.concat(server_name, Pool)
     namespace   = redis_namespace(server_name)
     node_ref    = :crypto.strong_rand_bytes(24)
-    node_name   = opts[:node_name]
+    node_name   = opts[:node_name] || raise ArgumentError, ":node_name is required option"
```
Comment (chrismccord) [inline lib/phoenix_pubsub_redis/redis.ex]: is *a require option

**Comment by chrismccord** [comment]
:shipit:

**Comment by chrismccord** [comment]
@hamiltop It feels odd to rely on `node()` being set properly when the user is not using distributed erlang, so I'm not sure how I feel about it. If we went that route in the instructions, we would at least need to detect if `node() == :"nonode@nohost"` and raise because we would have obvious dups

**Comment by hamiltop** [comment]
That's fair. BTW, https://github.com/phoenixframework/phoenix_pubsub_redis/issues/26 is opened with other presences issues.

**Comment by josevalim** [comment]
@chrismccord it is ok to ask for a name to be given but I would suggest using `--sname` and `--hidden`.

**Comment by chrismccord** [comment]
works for me. @mitchellhenke Can you amend with requiring a name, as follows:

1) first check for node(), and use the value so long as it's not `:nonode@nohost`, 2) next check opts[:node_name], 3) raise if both return no usuable value. Document usage of -sname as the default choice. Thanks!

**Comment by mitchellhenke** [comment]
Can do :)

**Comment by mitchellhenke** [comment]
@chrismccord forgot to ping after I added a commit from the feedback

**Comment by chrismccord** [comment]
<3 <3 <3

**Pull request merged**